dbt/include/athena/macros/materializations/snapshots/snapshot.sql

```diff
@@ -12,365 +11,112 @@
-              , {{ strategy.scd_id }} as dbt_scd_id
-            from snapshot_query
-        )
-        , upserts as (
-            select source_data.*
-              , case
-                  when snapshotted_data.dbt_unique_key IS NULL THEN 'insert'
-                  else 'update'
-                end as dbt_change_type
-              , {{ end_of_time() }} as dbt_valid_to
-              , True as is_current_record
-            from source_data
-            LEFT join snapshotted_data
-                   on snapshotted_data.dbt_unique_key = source_data.dbt_unique_key
-            where snapshotted_data.dbt_unique_key IS NULL
-               OR (
-                    snapshotted_data.dbt_unique_key IS NOT NULL
-                AND (
-                    {{ strategy.row_changed }}
-                )
-            )
-        )
-        {%- if strategy.invalidate_hard_deletes -%}
-          , deletes as (
-              select
-                  source_data.*
-                , 'delete' as dbt_change_type
-                , {{ end_of_time() }} as dbt_valid_to
-                , True as is_current_record
-              from snapshotted_data
-              LEFT join source_data
-                     on snapshotted_data.dbt_unique_key = source_data.dbt_unique_key
-              where source_data.dbt_unique_key IS NULL
-            )
-          select * from upserts
-          union all
-          select * from deletes;
-        {% else %}
-        select * from upserts;
-        {% endif %}
-    {% endif %}
-{%- endmacro %}
-
-{%
-  Create a new temporary table to hold the records to be upserted or deleted
-%}
-
-{% macro athena__build_snapshot_staging_table(strategy, sql, target_relation, table_type) %}
-    {% if table_type=='iceberg' %}
-        {% set temp_relation = make_temp_relation(target_relation) %}
-
-        {% set select = snapshot_staging_table(strategy, sql, target_relation, table_type) %}
-
-        {% call statement('build_snapshot_staging_relation') %}
-            {{ create_table_as(True, temp_relation, select) }}
-        {% endcall %}
-
-        {% do return(temp_relation) %}
-    {%else%}
-        {%- set tmp_identifier = target_relation.identifier ~ '__dbt_tmp' -%}
-
-        {%- set tmp_relation = api.Relation.create(identifier=tmp_identifier,
-                                                      schema=target_relation.schema,
-                                                      database=target_relation.database,
-                                                      type='table') -%}
-
-        {% do adapter.drop_relation(tmp_relation) %}
-
-        {%- set select = snapshot_staging_table(strategy, sql, target_relation, table_type) -%}
-
-        {% call statement('build_snapshot_staging_relation') %}
-            {{ create_table_as(False, tmp_relation, select) }}
-        {% endcall %}
-
-        {% do return(tmp_relation) %}
-    {% endif %}
+{% macro iceberg_snapshot_merge_sql(target, source, insert_cols) %}
```

Comment:
The original `athena__snapshot_merge_sql` macro is split in two separate macros, one for hive and one for iceberg. This is the iceberg part of that function without any other changes.

Reply:
It was not possible to use the original dbt-core merge sql, because Athena requires an alias for the source table.